test/create-sys-script.py

```diff
@@ -0,0 +1,202 @@
+
+
+def handle_file(outfile, path):
+    m = mode_rwx(os.lstat(path).st_mode)
+    f = open(path, "rb")
+    # read in as binary
+    b = f.read()
+    f.close()
+    # convert to string containing b'' representation
+    s = repr(b)
+    # strip leading b' and trailing '
+    s = s[2:-1]
+    # b'' contain newlines as \n, so convert back
+    s = s.replace(u'\\n', u'\n')
+    # we're placing it into a r'''''' so replace '
+    s = s.replace(u"'", u'\x27')
+    outfile.write(u"f(u'{}', u'{:o}', r'''{}''')\n".format(path, m, s))
+
+
+def process_sysdir(outfile):
+    for (dirpath, dirnames, filenames) in os.walk("sys"):
+        handle_dir(outfile, dirpath)
+        for d in dirnames:
+            path = os.path.join(dirpath, d)
+            if stat.S_ISLNK(os.lstat(path).st_mode):
+                handle_link(outfile, path)
+        for f in filenames:
+            path = os.path.join(dirpath, f)
+            mode = os.lstat(path).st_mode
+            if stat.S_ISLNK(mode):
+                handle_link(outfile, path)
+            elif stat.S_ISREG(mode):
+                handle_file(outfile, path)
+
+
+def verify_dir(tmpd, path_a):
+    path_b = os.path.join(tmpd, path_a)
+    mode_a = os.lstat(path_a).st_mode
+    mode_b = os.lstat(path_b).st_mode
+    if not stat.S_ISDIR(mode_b):
+        raise Exception(u"Not directory")
+    if mode_rwx(mode_a) != mode_rwx(mode_b):
+        raise Exception(u"Permissions mismatch")
+
+
+def verify_link(tmpd, path_a):
+    path_b = os.path.join(tmpd, path_a)
+    if not stat.S_ISLNK(os.lstat(path_b).st_mode):
+        raise Exception(u"Not symlink")
+    if os.readlink(path_a) != os.readlink(path_b):
+        raise Exception(u"Symlink dest mismatch")
+
+
+def verify_file(tmpd, path_a):
+    path_b = os.path.join(tmpd, path_a)
+    mode_a = os.lstat(path_a).st_mode
+    mode_b = os.lstat(path_b).st_mode
+    if not stat.S_ISREG(mode_b):
+        raise Exception(u"Not file")
+    if mode_rwx(mode_a) != mode_rwx(mode_b):
+        raise Exception(u"Permissions mismatch")
+    if not filecmp.cmp(path_a, path_b, shallow=False):
+        raise Exception(u"File contents mismatch")
+
+
+def verify_script(tmpd):
+    path = ""
+    try:
+        for (dirpath, dirnames, filenames) in os.walk("sys"):
+            path = dirpath
+            verify_dir(tmpd, path)
+            for d in dirnames:
+                path = os.path.join(dirpath, d)
+                if stat.S_ISLNK(os.lstat(path).st_mode):
+                    verify_link(tmpd, path)
+                for f in filenames:
+                    path = os.path.join(dirpath, f)
+                    mode = os.lstat(path).st_mode
+                    if stat.S_ISLNK(mode):
+                        verify_link(tmpd, path)
+                    elif stat.S_ISREG(mode):
+                        verify_file(tmpd, path)
+    except Exception as e:
+        print("FAIL on '{}'".format(path))
+        print(e)
+        exit(1)
```

Comment:
No need to catch the value of the exception:
```
try:
    ...
except Exception:
   print("FAIL ...", file=sys.stderr)
   raise
```
(It must print to stderr, otherwise the message will not be always visible.)